implementation/src/main/java/io/smallrye/mutiny/Uni.java

```diff
@@ -381,13 +382,25 @@ static UniCombine combine() {
      */
     Uni<T> runSubscriptionOn(Executor executor);
 
+    /**
+     * Configure memoization of the {@link Uni} item or failure.
+     * 
+     * @return the object to configure memoization
+     */
+    @Experimental("Memoization is an experimental feature at this stage")
+    UniMemoize<T> memoize();
+
     /**
      * Caches the events (item or failure) of this {@link Uni} and replays it for all further {@link UniSubscriber}.
      *
      * @return the new {@link Uni}. Unlike regular {@link Uni}, re-subscribing to this {@link Uni} does not re-compute
      *         the outcome but replayed the cached events.
+     * @deprecated Use {@link UniMemoize#indefinitely()} instead
      */
-    Uni<T> cache();
+    @Deprecated
+    default Uni<T> cache() {
```

Comment:
very funny: existing users will get a deprecation as well as an experimental warning at the same time 🤔

Reply:
> very funny: existing users will get a deprecation as well as an experimental warning at the same time 🤔

😂😂😂

Reply:
From now on, I'm going to mark all my APIs `@Experimental @Deprecated`.

Reply:
We need to introduce a `@EventuallyDeprecated` meta-annotation @gavinking

Reply:
`@Preprecated`